flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/JobResultStore.java

```diff
@@ -43,69 +44,70 @@ public interface JobResultStore {
      * Registers the passed {@link JobResultEntry} instance as {@code dirty} which indicates that
      * clean-up operations still need to be performed. Once the job resource cleanup has been
      * finalized, we can mark the {@code JobResultEntry} as {@code clean} result using {@link
-     * #markResultAsClean(JobID)}.
+     * #markResultAsCleanAsync(JobID)}.
      *
      * @param jobResultEntry The job result we wish to persist.
-     * @throws IOException if the creation of the dirty result failed for IO reasons.
-     * @throws IllegalStateException if the passed {@code jobResultEntry} has a {@code JobID}
-     *     attached that is already registered in this {@code JobResultStore}.
+     * @return a successfully completed future with {@code true} if the dirty result is created
+     *     successfully. The method will throw {@link IllegalStateException} if the passed {@code
+     *     jobResultEntry} has a {@code JobID} attached that is already registered in this {@code
+     *     JobResultStore}.
      */
-    void createDirtyResult(JobResultEntry jobResultEntry) throws IOException, IllegalStateException;
+    CompletableFuture<Void> createDirtyResultAsync(JobResultEntry jobResultEntry);
 
     /**
      * Marks an existing {@link JobResultEntry} as {@code clean}. This indicates that no more
      * resource cleanup steps need to be performed. No actions should be triggered if the passed
      * {@code JobID} belongs to a job that was already marked as clean.
      *
      * @param jobId Ident of the job we wish to mark as clean.
-     * @throws IOException if marking the {@code dirty} {@code JobResultEntry} as {@code clean}
-     *     failed for IO reasons.
-     * @throws NoSuchElementException if there is no corresponding {@code dirty} job present in the
-     *     store for the given {@code JobID}.
+     * @return a successfully completed future if the result is marked successfully, The future will
+     *     completed with {@link NoSuchElementException} if there is no corresponding {@code dirty}
+     *     job present in the store for the given {@code JobID}.
```

Comment:
We should remove the documentation for the `NoSuchElementException`. That should be still part of the contract and, therefore, mentioned in the JavaDoc. Could you add this as part of the returned future documentation as an additional sentence?

Reply:
Fixed.